aibolit/ast_framework/java_class_decomposition.py

```diff
@@ -41,31 +56,54 @@ def is_ast_pattern(class_ast: AST, Pattern) -> bool:
     return len(Pattern().value(class_ast)) > 0
 
 
+def _normalize_strength(strength: Union[DecompositionStrength, str]) -> DecompositionStrength:
+    if isinstance(strength, DecompositionStrength):
+        return strength
+    if isinstance(strength, str):
+        try:
+            return DecompositionStrength(strength.lower())
+        except ValueError as exc:
+            valid_strengths = DecompositionStrength.values()
+            raise ValueError(
+                f'Unsupported decomposition strength: {strength}. '
+                f'Must be one of: {valid_strengths}'
+            ) from exc
+    valid_strengths = DecompositionStrength.values()
+    raise ValueError(
+        f'Unsupported decomposition strength: {strength}. '
+        f'Must be one of: {valid_strengths}'
+    )
+
+
 def decompose_java_class(
         class_ast: AST,
-        strength: str,
+        strength: Union[DecompositionStrength, str] = DecompositionStrength.STRONG,
         ignore_setters=False,
         ignore_getters=False) -> List[AST]:
-    '''
+    """
     Splits java_class fields and methods by their usage and
     construct for each case an AST with only those fields and methods kept.
     :param class_ast: component
     :param ignore_getters: should ignore getters
     :param ignore_setters: should ignore setters
-    :param strength: controls splitting criteria. Use "strong" or "weak"
+    :param strength: decomposition strength; accepts DecompositionStrength or 'strong'/'weak'
     for splitting fields and methods by strong and weak connectivity.
-    '''
+    """
 
     usage_graph = _create_usage_graph(class_ast)
 
     components: Iterator[Set[int]]
-    if strength == 'strong':
+    normalized_strength = _normalize_strength(strength)
+
+    if normalized_strength == DecompositionStrength.STRONG:
         components = strongly_connected_components(usage_graph)
-    elif strength == 'weak':
+    elif normalized_strength == DecompositionStrength.WEAK:
         components = weakly_connected_components(usage_graph)
     else:
+        valid_strengths = [s.value for s in DecompositionStrength]
         raise ValueError(
-            f"'strength' argument must be either 'strong' or 'weak', but '{strength}' was provided."
+            f'Unsupported decomposition strength: {normalized_strength}. '
```

Comment:
I suppose there is no need to raise error here, as `_normalize_strength` already raises the very same exception.